lib/ecommerce/request.rb

```diff
@@ -0,0 +1,49 @@
+module Ecommerce
+  class Request
+
+    def initialize(args)
+      @args = args
+    end
+
+    def run
+      request.run
+      request.response
+    end
+
+    private
+
+    attr_reader :args
+
+    def request
+      @request ||= Typhoeus::Request.new(args[:url], options)
+    end
+
+    def options
+      {
+        method:           args[:method],
+        params:           args[:params],
+        body:             body,
+        headers:          headers,
+        accept_encoding:  "gzip"
+      }.reject {|k,v| v.nil?}
```

Comment:
Unused block argument - `k`. If it's necessary, use `_` or `_k` as an argument name to indicate that it won't be used.<br>Space missing after comma.<br>Space between { and | missing.<br>Space missing inside }.